app/src/main/java/com/jwisozk/flightreviews/ReviewAdapter.kt

```diff
@@ -0,0 +1,131 @@
+        fun bind(@NonNull itemView: View, position: Int) {
+            if (position == penultimateIndex) {
+                val checkBox = itemView.findViewById<CheckBox>(R.id.foodCheckBox)
+                checkBox.visibility = View.VISIBLE
+                checkBox.setOnCheckedChangeListener { _, isChecked ->
+                    if (isChecked) viewModel.setParameters(foodParam = null)
+                }
+            }
+            val ratingBar = itemView.findViewById<RatingBar>(R.id.ratingBar)
+            ratingBar.onRatingBarChangeListener =
+                RatingBar.OnRatingBarChangeListener { _, rating, _ ->
+                    run {
+                        when (position) {
+                            0 -> viewModel.onOverallRatingChanged(rating, ParamEnum.PEOPLE)
+                            1 -> viewModel.onOverallRatingChanged(rating, ParamEnum.AIRCRAFT)
+                            2 -> viewModel.onOverallRatingChanged(rating, ParamEnum.SEAT)
+                            3 -> viewModel.onOverallRatingChanged(rating, ParamEnum.CREW)
+                            4 -> {
+                                param?.food?.let {
+                                    viewModel.onOverallRatingChanged(rating, ParamEnum.FOOD)
+                                }
+
+                            }
+                        }
+                    }
+                }
+            val textView = itemView.findViewById<TextView>(R.id.labelRatingBar)
+            textView.text = labelArray[position]
+        }
+    }
+
+    inner class EditTextHolder(@NonNull itemView: ViewGroup) : RecyclerView.ViewHolder(itemView) {
+        fun bind(@NonNull itemView: View, position: Int) {
+            editText = itemView.findViewById(R.id.feedbackEditText)
+            val textView = itemView.findViewById<TextView>(R.id.labelEditText)
+            textView.text = labelArray[position]
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return when (position) {
+            labelArray.size -> R.layout.button
+            lastIndex -> R.layout.edit_text
+            else -> R.layout.rating_bar
+        }
+    }
+
+    @NonNull
+    override fun onCreateViewHolder(
+        @NonNull parent: ViewGroup,
+        viewType: Int
+    ): RecyclerView.ViewHolder {
+        return when (viewType) {
+            R.layout.rating_bar -> {
+                RatingBarHolder(LayoutInflater.from(parent.context)
+                    .inflate(R.layout.rating_bar, parent, false) as ViewGroup)
+            }
+            R.layout.edit_text -> {
+                EditTextHolder(LayoutInflater.from(parent.context)
+                .inflate(R.layout.edit_text, parent, false) as ViewGroup)
+            }
+            else -> {
+                ButtonHolder(LayoutInflater.from(parent.context)
+                    .inflate(R.layout.button, parent, false))
+            }
```

Comment:
Let's do inflation in holders classes like `super(layoutInflater.inflate(...)`, and for this purpose holders should take 
```kotlin
inflater: LayoutInflater,
 parent: ViewGroup,
```
as arguments